### PR TITLE
main/firmware-linux: update to 20240115

### PR DESCRIPTION
**Pull request by nekopsykose**

### CODE REVIEW

File: main/firmware-linux/template.py [hunk 1]
```diff
@@ -274,6 +283,13 @@
         "network",
         ["adaptec/starfire*"],
     ),
+    (
+        "ti-tas",
+        "Texas Instruments Amplifiers",
+        None,
+        "audio",
+        ["INT88*", "TAS2*", "TIAS2*", "ti/tas*"],
+    ),
```
Comment (nekopsykose) [inline main/firmware-linux/template.py:292]: this one is annoying:
```
 INT8866RCA2.bin -> ti/tas2563/INT8866RCA2.bin
 TAS2XXX38A5.bin -> ti/tas2781/TAS2XXX38A5.bin
 TAS2XXX38A7.bin -> ti/tas2781/TAS2XXX38A7.bin
 TAS2XXX38A8.bin -> ti/tas2781/TAS2XXX38A8.bin
 TAS2XXX38B8.bin -> ti/tas2781/TAS2XXX38B8.bin
 TAS2XXX38B9.bin -> ti/tas2781/TAS2XXX38B9.bin
 TAS2XXX38BA.bin -> ti/tas2781/TAS2XXX38BA.bin
 TAS2XXX38BB.bin -> ti/tas2781/TAS2XXX38BB.bin
 TAS2XXX38BE.bin -> ti/tas2781/TAS2XXX38BE.bin
 TAS2XXX38BF.bin -> ti/tas2781/TAS2XXX38BF.bin
 TAS2XXX38C3.bin -> ti/tas2781/TAS2XXX38C3.bin
 TAS2XXX38CB.bin -> ti/tas2781/TAS2XXX38CB.bin
 TAS2XXX38CD.bin -> ti/tas2781/TAS2XXX38CD.bin
 TAS2XXX38D3.bin -> ti/tas2781/TAS2XXX38D3.bin
 TAS2XXX38D4.bin -> ti/tas2781/TAS2XXX38D4.bin
 TAS2XXX38D5.bin -> ti/tas2781/TAS2XXX38D5.bin
 TAS2XXX38D6.bin -> ti/tas2781/TAS2XXX38D6.bin
 TAS2XXX38DF.bin -> ti/tas2781/TAS2XXX38DF.bin
 TAS2XXX38E0.bin -> ti/tas2781/TAS2XXX38E0.bin
 TAS2XXX387D.bin -> ti/tas2781/TAS2XXX387D.bin
 TAS2XXX387E.bin -> ti/tas2781/TAS2XXX387E.bin
 TAS2XXX387F.bin -> ti/tas2781/TAS2XXX387F.bin
 TAS2XXX3870.bin -> ti/tas2563/TAS2XXX3870.bin
 TAS2XXX3880.bin -> ti/tas2781/TAS2XXX3880.bin
 TAS2XXX3881.bin -> ti/tas2781/TAS2XXX3881.bin
 TAS2XXX3882.bin -> ti/tas2781/TAS2XXX3882.bin
 TAS2XXX3884.bin -> ti/tas2781/TAS2XXX3884.bin
 TAS2XXX3886.bin -> ti/tas2781/TAS2XXX3886.bin
 ti/
 TIAS2781RCA2.bin -> ti/tas2781/TIAS2781RCA2.bin
 TIAS2781RCA4.bin -> ti/tas2781/TIAS2781RCA4.bin
```

there's technically `tas2563` and `tas2781`. but the 2563 name is just random numbers like the other, so i kept them together

File: main/firmware-linux/template.py [hunk 2]
```diff
@@ -107,6 +109,13 @@
     ),
     ("intel-bt", "Intel Bluetooth", _arch_x86, "network", ["intel/ibt*"]),
     ("intel-ice", "Intel E800 series", _arch_x86, "network", ["intel/ice"]),
+    (
+        "intel-ivsc",
+        "Intel Visual Sensing Controller",
+        _arch_x86,
```
Comment (nekopsykose) [inline main/firmware-linux/template.py:115]: this is new *lake only afaict so this should be accurate https://github.com/intel/ivsc-driver

also for reference https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/commit/?id=cfc997c664f076c6800b749ed7ef315b1e8e76fd

File: main/firmware-linux/template.py [hunk 3]
```diff
@@ -66,6 +67,7 @@
     ),
     ("cpia2", "STM CPIA2 cameras", None, "misc", ["cpia2"]),
     ("cs35l41", "Cirrus CS35L41 amplifier", None, "audio", ["cirrus/cs35l41*"]),
+    ("cs42l43", "Cirrus CS42L43 amplifier", None, "audio", ["cs42l43.bin"]),
```
Comment (nekopsykose) [inline main/firmware-linux/template.py:70]: why did they leave this as a random ass file..

File: main/firmware-linux/template.py [hunk 4]
```diff
@@ -283,6 +292,13 @@
         ["ti-connectivity"],
     ),
     ("ti-serial", "TI USB 3410/5052", None, "misc", ["ti_*.fw"]),
+    (
+        "ti-tas",
+        "Texas Instruments Amplifiers",
```
Comment (q66) [inline main/firmware-linux/template.py:297]: lowercase amplifiers i think?

Comment (nekopsykose) [reply main/firmware-linux/template.py:297]: yea, not sure why i titlecased it lol

**Pull request closed without merging**